### ISSUE TITLE
[Bug]: The most basic example doesn't work with matplotlib

### ISSUE THREAD

**Issue by MatthewCroughan** [tweag/jupyterWith#361]
### Current Behavior

If you follow the most basic example of "Extending Kernels" from [the documentation](https://jupyterwith.tweag.io/tutorials/#extending-kernels), adding `matplotlib` do the list of `extraPackages` does not work.

It looks like an override provided by poetry2nix is failing to apply. How has this not re-invented some of the the dependency hell problem that Nix is supposed to solve?

```
user: matthew ~/tmp/my-project
❯ cat kernels/python.nix
{
  name,
  availableKernels,
  extraArgs,
}:
availableKernels.python {
  inherit name;
  inherit (extraArgs) pkgs;
  displayName = "Custom ${name}";
  extraPackages = ps: [
    ps.numpy
    ps.matplotlib
  ];
}

user: matthew ~/tmp/my-project
❯ nix run
warning: Using saved setting for 'extra-substituters = https://tweag-jupyter.cachix.org' from ~/.local/share/nix/trusted-settings.json.
warning: Using saved setting for 'extra-trusted-public-keys = tweag-jupyter.cachix.org-1:UtNH4Zs6hVUFpFBTLaA4ejYavPo5EFFqgd7G7FxGW9g=' from ~/.local/share/nix/trusted-settings.json.
error: attribute 'format' missing

       at /nix/store/p6xfpp6i1a2h4ia3v81d1r414v6a603i-source/pkgs/development/tools/poetry2nix/poetry2nix/overrides/default.nix:1463:40:

         1462|             ++ lib.optionals (self.isPyPy) [ tk xorg.libX11 ];
         1463|           preConfigure = lib.optional (old.format != "wheel") preConfigure;
             |                                        ^
         1464|         }
(use '--show-trace' to show detailed location information)
```

### Expected Behavior

That matplotlib would successfully build.

### Steps To Reproduce

Add `ps.matplotlib` to the `extraPackages` argument of the `availableKernels.python` function, as described in the  documentation: https://jupyterwith.tweag.io/tutorials/#extending-kernels

### Version

```
user: matthew ~ 
❯ nix-info -m
 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.7-zen1, NixOS, 22.11 (Raccoon), 22.11.20221109.872fcee`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `""`
 - channels(matthew): `""`
 - nixpkgs: `/nix/store/y63kal4f9mmdy14mdfp8d24h6hfi4cqp-source`
```

**Comment by AmeerTaweel** [comment]
The same issue happens with `scikit-learn`...

**Comment by djacu** [comment]
This should be fixed with https://github.com/tweag/jupyterWith/pull/415.
A new example kernel was added to the CI that pulls in `numpy`, `scipy`, and `matplotlib` via `extraPackages`. All seems to be working now.

I tested this with a [scikit-learn demo](https://scikit-learn.org/stable/auto_examples/cluster/plot_kmeans_digits.html#sphx-glr-auto-examples-cluster-plot-kmeans-digits-py) and it worked locally. Give it another try and let me know if it works for you.
